doc/2/api/controllers/auth/create-api-key/index.md

```diff
@@ -0,0 +1,103 @@
+---
+code: true
+type: page
+title: createApiKey
+---
+
+# createApiKey
+
+Creates a new API key.
+
+---
+
+## Query Syntax
+
+### HTTP
+
+```http
+URL: http://kuzzle:7512/api-keys/_create[?refresh=wait_for][&expiresIn=-1][&_id=null]
+Method: POST
+Body:
+```
+
+```js
+{
+  "description": "Sigfox callback authentication api key"
+}
+```
+
+### Other protocols
+
+```js
+{
+  "controller": "auth",
+  "action": "createApiKey",
+  "body": {
+    "description": "Sigfox callback authentication api key"
+  },
+
+  // optional arguments
+  "_id": "api-key-id",
+  "expiresIn": -1,
+  "refresh": "wait_for"
+}
+```
+
+---
+
+## Arguments
+  
+### Optional:
+
+- `expiresIn`: set the expiration duration (`-1` by default)
+  - if a raw number is provided (not enclosed between quotes), then the expiration delay is in milliseconds. Example: `86400000`
+  - if this value is a string, then its content is parsed by the [ms](https://www.npmjs.com/package/ms) library. Examples: `"6d"`, `"10h"`
+  - if `-1` is provided, the token will never expire
+- `refresh`: if set to `wait_for`, Kuzzle will not respond until the newly created API key is indexed
```

Comment:
Maybe we can set the option to `wait_for` by default? 
An administrator may want to create batch of API keys for example.

Also, it's not a very strong argument but it allows to speed up functional test by not waiting ES indexation each time we create an API key

Reply:
I've worked just yesterday on the subject, putting more `wait_for` options to security routes used by functional tests, because those became instable thanks to race conditions. 
So I'm not sure about your functional test argument :grin: 

But why not making `refresh = wait_for` by default. Should be the case for all security operations really.

Reply:
I use it only when I have to do some search, otherwise I try to work only with ID :sweat_smile: 

Yes I agree, we should modify all securities action with this